api/database/migrations/permissions.go

```diff
@@ -0,0 +1,154 @@
+package migrations
+
+import (
+	"github.com/photoview/photoview/api/graphql/auth"
+	"github.com/photoview/photoview/api/graphql/models"
+	"github.com/pkg/errors"
+	"gorm.io/gorm"
+)
+
+// MigrateForPermissions Migrates new permissions into the system and upgrades system roles according to the new permissions.
+func MigrateForPermissions(db *gorm.DB) error {
```

Comment:
We should test this migration thoroughly before releasing, since it is a little more involved.
Just to make sure a lot of users databases doesn't get corrupted.

Reply:
Fully agree on this one, I've tested the scenarios I can think of albeit on a small dataset and only a few user's. These are as below.
- Migrating from an old version of photoview with various admin users, normal user & demo user
- Addition of a new permission
- Addition of an existing permission to a system user group doesn't add to the user group
- Addition of a new permission to a system user group automatically adds to the user group